### ISSUE TITLE
Running out of memory when checking out files.

### ISSUE THREAD

**Issue by tumi23** [SRombauts/UE4PlasticPlugin#97]
Hi

I've been using the plugin for some time and have had problems with checking out files(happens a lot in world partition) where I check in files in the editor and then at the end starts for some reason reverting said files that I'm just checking in and while that is happening if I look at the task manager the memory of the UnrealEditor.exe keeps on getting bigger and bigger until it runs out of memory.

**Comment by tumi23** [comment]
After further testing it seems to be related to deleting a large amount of files, It starts by checking it out then reverting them and finally deleting them. The problem seems to be that the reverting process stores the files in memory in if too many files are deleted the entire memory is consumed and the editor crashes.

More info: I have 32GB's of RAM and when deleting maybe...150+ files and saving them leads to the entire memory being consumed. Seems to be some problem with storing the assets in memory when reverting them which ends up consuming all memory.

**Comment by SRombauts** [comment]
Hi @tumi23 

Thanks for reporting the issue!
Could you let me know what version of the plugin, and with what version of Unreal you are using it?
If you could post or send me the corresponding Unreal logs?

(side question, probably irrelevant: Do you have a dedicated GPU ? or is this a laptop using shared memory from your system? We had some report of crash out of video memory I could never repro)

**Comment by SRombauts** [comment]
See https://github.com/SRombauts/UE4PlasticPlugin/issues/93

**Comment by SRombauts** [comment]
Thanks, that will help for sure!
It seems that there are quite some trouble with One File Per Actor, I'll have to investigate more thoroughly as soon as I am done with the last batch of changes needed for UE5

This behaviour of delete that first checkout is new to me, I'll have to also double check this, might be because of UE5 or something with OFPA...

**Comment by RoadkilladilloStudio** [comment]
Guess I better report this too is still happening on v1.6.0 ...I have to work in DX12 as well and it appears if you need to move a folder with a lot of assets you still get the Out of Video Memory crash. I am only with a 2080ti on my working setup but just watched it and it appears it overloads the Video memory to max within a few minutes of trying to relocate a folder I have. Noticed also my RAM itself will scale up to about half of what my system has. 64gbs in there but it makes it to around 32gbs when this crash happens.

Figured pass it on in case its expected on this latest plugin version to work. Today I am having to do some organize and so far 4 folders have crashed. I guess for now I can just try to move less at a time but really slows things down there

**Comment by SRombauts** [comment]
Thanks a lot, I have been trying to investigate a bit more yesterday, but even trying to checkout ~1800 files from the cloud didn't break, it, though it was quite fast
It might depend on the speed of it, eg the Engine or plugin allocating memory per frame, so the longer it takes, the worst it would get

**Comment by SRombauts** [comment]
This should now be worked around on master with https://github.com/SRombauts/UE4PlasticPlugin/commit/974ff8fbbefd7c95ef157f2ef04f70007888724e

**Comment by SRombautsU** [comment]
I have started reproducing the issue in Unreal without any Source Control operation, with a simple infinite loop in the Provider, meaning it's purely a render thread issue (and I can see how other progress bars in the Editor code seem to handle this)
<img width="892" alt="2022-08-17 (2)" src="https://user-images.githubusercontent.com/101874327/185143024-640a2ac4-017f-4b82-93e9-a03848e4cc77.png">

**Comment by SRombautsU** [comment]
For future me searching the info, it's tracked by the oldest issue #86 
```
[SRombauts](https://github.com/SRombauts) commented [21 days ago](https://github.com/SRombauts/UEPlasticPlugin/issues/86#issuecomment-1218373488) • 
I opened a PR for Epic Games to integrate the proper fix into Unreal Engine 5.1 hopefully:
https://github.com/EpicGames/UnrealEngine/pull/9492 [UE-100291][UE-160299] Fix source control progress bar crashing out of video memory in case of long-running operation #9492

[SRombauts](https://github.com/SRombauts) commented [13 days ago](https://github.com/SRombauts/UEPlasticPlugin/issues/86#issuecomment-1227601539)
The proper fix has been submitted to ue5-main https://github.com/EpicGames/UnrealEngine/commit/67dacf2baaafe0d9b6fb5eeb5b179fb5b871a4bf

It will be part of UE 5.1 release :)
```

**Comment by SRombauts** [comment]
For future me searching the info, it's tracked by the oldest issue #86 
```
[SRombauts](https://github.com/SRombauts) commented [21 days ago](https://github.com/SRombauts/UEPlasticPlugin/issues/86#issuecomment-1218373488) • 
I opened a PR for Epic Games to integrate the proper fix into Unreal Engine 5.1 hopefully:
https://github.com/EpicGames/UnrealEngine/pull/9492 [UE-100291][UE-160299] Fix source control progress bar crashing out of video memory in case of long-running operation #9492

[SRombauts](https://github.com/SRombauts) commented [13 days ago](https://github.com/SRombauts/UEPlasticPlugin/issues/86#issuecomment-1227601539)
The proper fix has been submitted to ue5-main https://github.com/EpicGames/UnrealEngine/commit/67dacf2baaafe0d9b6fb5eeb5b179fb5b871a4bf

It will be part of UE 5.1 release :)
```
